patches/0037-Non-idempotent-sns-topic-creation.patch

```diff
@@ -0,0 +1,64 @@
+diff --git a/internal/service/sns/topic.go b/internal/service/sns/topic.go
+index 98ebd87d5d..adc13a9f58 100644
+--- a/internal/service/sns/topic.go
++++ b/internal/service/sns/topic.go
+@@ -9,6 +9,7 @@ import (
+ 	"fmt"
+ 	"log"
+ 	"regexp"
++	"sync"
+ 	"time"
+ 
+ 	"github.com/YakDriver/regexache"
+@@ -259,6 +260,8 @@ func resourceTopic() *schema.Resource {
+ 	}
+ }
+ 
++var snsGlobalMutex sync.Mutex
++
+ func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+ 	var diags diag.Diagnostics
+ 	conn := meta.(*conns.AWSClient).SNSClient(ctx)
+@@ -283,6 +286,34 @@ func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta inter
+ 		delete(attributes, topicAttributeNameFIFOTopic)
+ 	}
+ 
++	// Look up if the topic already exists
++	snsGlobalMutex.Lock()
++	var next_token *string = nil
++	for done := false; !done; {
++		output, err := conn.ListTopics(ctx, &sns.ListTopicsInput{
++			NextToken: next_token,
++		})
++		if err != nil {
++			return diag.FromErr(err)
++		}
++
++		for _, topic := range output.Topics {
++			// split the topic arn to get the name
++			arn, err := arn.Parse(*topic.TopicArn)
++			if err != nil {
++				return diag.FromErr(err)
++			}
++			if arn.Resource == name {
++				return diag.Errorf("SNS Topic (%s) already exists", name)
++			}
++		}
++
++		if output.NextToken == nil {
++			done = true
++		}
++		next_token = output.NextToken
++	}
++
+ 	output, err := conn.CreateTopic(ctx, input)
+ 
+ 	// Some partitions (e.g. ISO) may not support tag-on-create.
+@@ -291,6 +322,7 @@ func resourceTopicCreate(ctx context.Context, d *schema.ResourceData, meta inter
+ 
+ 		output, err = conn.CreateTopic(ctx, input)
```

Comment:
What does conn.CreateTopic return if the topic already exists?

Reply:
Here is what the docs say:
```
This action is idempotent, so if the requester already owns a topic with the specified name, that topic's ARN is returned without creating a new topic.
```

And in my testing that seems to indeed be the behaviour.

Reply:
Neither our docs nor terraforms mention anything about the create returning existing resources:
https://www.pulumi.com/registry/packages/aws/api-docs/sns/topic/
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/sns_topic

Given that the new behaviour is the default behaviour for pulumi resources, I'd say this does not warrant any additional documentation, LMK if you disagree